doc: Update Istio GSG for Istio 0.7.0 and sidecar HTTP proxy

rlenglet opened the pull request:
Update the GSG to run Cilium with Istio using Docker images built from https://github.com/cilium/istio/tree/inject-cilium-filters and the Envoy binary built from https://github.com/cilium/cilium/pull/3035.

Signed-off-by: Romain Lenglet <romain@covalent.io>

rlenglet committed:
doc: Update Istio GSG for Istio 0.7.0

Signed-off-by: Romain Lenglet <romain@covalent.io>

tgraf commented on examples/kubernetes-istio/cilium-kube-inject.sed:
```diff
@@ -0,0 +1,13 @@
+# allow Cilium's Envoy filters to communicate with the Cilium agent.
+#
+# volumeMounts:
+# - mountPath: /var/run/cilium
```
We'll want to share a subdirectory in the future so we don't have to expose the entire state directory with each sidecar.

raybejjani replied:
Is it possible to mount the .sock directly? I've seen it done with docker but I don't know if volumeMounts/volumes allows it.

rlenglet replied:
Maybe. I’ll look into that. I have 2 sockets to mount. I went with the simplest solution for now.

tgraf commented on examples/kubernetes/cilium-sidecar.yaml:
```diff
@@ -0,0 +1,292 @@
+          - "--kvstore-opt"
+          - "etcd.config=/var/lib/etcd-config/etcd.config"
+          - "--disable-ipv4=$(DISABLE_IPV4)"
+          - "--sidecar-http-proxy"
```
In future, we could make this `--sidecar-http-proxy=$(SIDECAR_MODE)" so we can use a single DaemonSet and only provide a different configmap for the sidecar mode.

rlenglet replied:
The big plan is to make this setting per-endpoint at some point, so I didn't make this configuration polished.

raybejjani commented on examples/kubernetes-istio/bookinfo-details-v1.yaml:
```diff
@@ -49,7 +49,7 @@ specs:
           http:
           - method: GET
-            path: "^/details/[0-9]*$"
+            path: "/details/[0-9]*"
```
Out of curiosity, are the patterns auto-anchored or might this match anything that ends with `/details/[0-9]*`?

rlenglet replied:
With Envoy they’re supposed to be.

aanm commented on examples/kubernetes/cilium-sidecar.yaml:
```diff
@@ -0,0 +1,292 @@
+kind: ConfigMap
```
Can we wait until this PR is merged https://github.com/cilium/cilium/pull/3125? that way we can move this file to the `plugins` / `addons` directory in `examples/kubernetes`

rlenglet replied:
Sure, np.

rlenglet committed:
examples/kubernetes: Generate daemon sets defs for sidecar mode

Clean up examples/kubernetes/Makefile. Add support for V=0 quiet
option. Add support for multiple daemon set definitions for each
Kubernetes version.

Generate */cilium-sidecar*.yaml daemon set files for running
Cilium along with Istio.

Signed-off-by: Romain Lenglet <romain@covalent.io>

aanm reviewed (dismissed):
@rlenglet Can you remove the `examples/cilium-sidecar.yaml` and update the documentation with something like

```
.. tabs::
  .. group-tab:: K8s 1.7

    .. parsed-literal::

      $ wget \ |SCM_WEB|\/examples/kubernetes/1.7/cilium.yaml
      $ vim cilium.yaml
      [adjust the etcd address]

  .. group-tab:: K8s 1.8

    .. parsed-literal::

      $ wget \ |SCM_WEB|\/examples/kubernetes/1.8/cilium.yaml
      $ vim cilium.yaml
      [adjust the etcd address]

  .. group-tab:: K8s 1.9

    .. parsed-literal::

      $ wget \ |SCM_WEB|\/examples/kubernetes/1.9/cilium.yaml
      $ vim cilium.yaml
      [adjust the etcd address]

  .. group-tab:: K8s 1.10

    .. parsed-literal::

      $ wget \ |SCM_WEB|\/examples/kubernetes/1.10/cilium.yaml
      $ vim cilium.yaml
      [adjust the etcd address]

  .. group-tab:: K8s 1.11

    .. parsed-literal::

      $ wget \ |SCM_WEB|\/examples/kubernetes/1.11/cilium.yaml
      $ vim cilium.yaml
      [adjust the etcd address]
```

manalibhutiyani commented on Documentation/gettingstarted/istio.rst:
```diff
@@ -78,42 +78,76 @@ including ``kube-dns``, run:
+    clusterrole "cilium" created
+
+Kubernetes is now deploying Cilium with its RBAC settings, ConfigMap
+and DaemonSet as a pod on minkube. This operation is performed in the
```
Nit: `minkube` --> `minikube`

rlenglet replied:
Fixed.

manalibhutiyani commented on examples/kubernetes-istio/cilium-kube-inject.sed:
```diff
@@ -0,0 +1,13 @@
+# Inject Cilium's /var/run/cilium directory into each Istio sidecar proxy to
```
Nit: I think `mount` is the correct word and not `inject` a directory?

rlenglet replied:
I prefer to use the "inject" term because this is in the context of running the `istioctl kube-inject` commands, which modifies the deployment specs to _inject_ new containers, labels, etc. So "inject" seems like the proper term.

rlenglet committed:
doc: Use K8s-version-specific YAML files in Istio GSG

Signed-off-by: Romain Lenglet <romain@covalent.io>
